crates/astria-composer/src/searcher/mod.rs

```diff
@@ -154,24 +124,52 @@ impl Searcher {
     /// + failed to connect to the eth RPC server;
     /// + failed to construct a sequencer clinet
     pub(super) async fn from_config(cfg: &Config) -> eyre::Result<Self> {
+        use futures::{
+            FutureExt as _,
+            StreamExt as _,
+        };
         use rollup::Rollup;
-        let mut rollups = cfg
+        let rollups = cfg
             .rollups
             .split(',')
             .filter(|s| !s.is_empty())
             .map(|s| Rollup::parse(s).map(Rollup::into_parts))
             .collect::<Result<HashMap<_, _>, _>>()
             .wrap_err("failed parsing provided <chain_id>::<url> pairs as rollups")?;
-        ensure!(
-            rollups.len() == 1,
-            "currently only 1 rollup is supported; got {}",
-            rollups.len(),
-        );
-        // connect to eth node
-        let (rollup_chain_id, execution_url) = rollups.drain().next().unwrap();
-        let eth_client = EthClient::connect(&execution_url)
-            .await
-            .wrap_err("failed connecting to eth")?;
+
+        let (tx_sender, new_transactions) = mpsc::channel(256);
+
+        let mut create_collectors = rollups
+            .into_iter()
+            .map(|(chain_id, url)| {
+                let task_name = chain_id.clone();
+                tokio::spawn(Collector::new(chain_id, url, tx_sender.clone()))
+                    .map(move |result| (task_name, result))
+            })
```

Comment:
why start all the collectors in tasks instead of just synchronously?

Reply:
Because then we would have to wait for the sum of their startup times. This way it only takes as long as the slowest. It's just better. :-)